pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformContext.java

```diff
@@ -182,10 +181,21 @@ public static byte[] serializeGenericRecord(GenericRecord record) throws IOExcep
     return oo.toByteArray();
   }
 
-  public void addOrReplaceAvroValueFields(
+  public void addOrReplaceValueFields(
+      SchemaType schemaType,
+      Map<org.apache.avro.Schema.Field, Object> newFields,
+      Map<org.apache.avro.Schema, org.apache.avro.Schema> schemaCache) {
+    if (schemaType == SchemaType.AVRO) {
+      addOrReplaceAvroValueFields(newFields, schemaCache);
+    } else if (schemaType == SchemaType.JSON) {
+      addOrReplaceJsonValueFields(newFields, schemaCache);
+    }
```

Comment:
fail ?

Reply:
I don't know. We should either fail or ignore.
I'm not sure we're consistent everywhere for this.